### DISCUSSION TITLE
outputStandalone - allow adding implicit application dependencies in the output

### DISCUSSION THREAD

**Discussion by triartleet** [vercel/next.js#38004] [Ideas]
### Describe the feature you'd like to request

**Before** running my application, I need to _preload_ a local module which instruments my app code and customizes some low-level modules which Next uses (e.g.`node -r ./preloadModule.js server.js`). To add, this local module depends on some packages which are not used by the application. Preloading the local module in regular builds works since `node_modules` includes any package. 

This won't work with `standaloneOutput` since the preloaded module's dependencies are not used by any application file.

For a more detailed description of the setup, you can read this [post](https://jake.tl/notes/2021-04-04-nextjs-preload-hack). The post utilizes a [server-preload.js](https://gist.github.com/justjake/6fa39b07421a58ad58ff013b141b7172#file-server-preload-js) which is very similar to my implementation.

### Describe the solution you'd like

If the feature request IS relevant to its scope, I would like a solution that could satisfy the aforementioned use case without adding overhead (custom solutions) or compromisations (disabling features).

Examples
* Would it be possible for example to add our own tracing sources? 
* Could the default server configuration be extended / or at least have the custom server part of the standalone output? I found some discussion for the former as in https://github.com/vercel/next.js/discussions/11686 .

### Describe alternatives you've considered

- I tried to bundle my preload module separately to keep it decoupled. My POC did not work yet though. 
- I probably can cheat with adding the extra dependencies by importing them in some api/route but this seems very hackish.

**Any suggestions are highly appreciated!**

**Comment by izi-p** [comment]
Another use-case where this can be needed as well:

We use a custom server that uses some dependencies.. (express and stuff..)
This server is written in ts, it is not bundled with its dependencies. It's just transpiled to js.
When we build in normal mode, these dependencies ends up in the node_modules and everything works fine.

When we build in standalone mode, these dependencies are cleaned in some ways. The custom-server can't import these dependencies. The solution I have found is to properly build and bundle the custom server on it's own with its dependencies but it would be better to explicitely specify these dependencies and to just transpile custom-server files.

**Reply by GoldenTK** [reply]
Did you try to run output „standalone” with custom server?

I have to use nextjs package separetly because standalone (lightweight) nextjs package do not include next/server dependencies :(.
